browser-extension/src/models/banner/Banner.js

```diff
@@ -0,0 +1,156 @@
+import { isNavigatorLang } from '../../utils.js';
+import { defaultBannerOptions } from './BannerOptions.js';
+
+export class Banner {
```

Comment:
This `Banner` is not generic yet, mainly because of the specific translation keys.

Let's add comment so that we can make this class generic later.

```suggestion
/**
 * A banner displaying a message and an action button.
 * 
 * At the moment this class is specific to the extension research study, but
 * could be refactored to be generic.
 * 
 * - the texts could be parameters of the constructor
 * - the action button could be optional
 * - etc.
 */
export class Banner {
```